wrappers/desktop.go

```diff
@@ -103,8 +104,9 @@ var isValidDesktopFileLine = regexp.MustCompile(strings.Join([]string{
 	"^TargetEnvironment=",
 }, "|")).Match
 
-// rewriteExecLine rewrites a "Exec=" line to use the wrapper path for snap application.
-func rewriteExecLine(s *snap.Info, desktopFile, line string) (string, error) {
+// detectAppAndRewriteExecLine parses snap app from passed "Exec=" line and rewrites it
+// to use the wrapper path for snap application.
+func detectAppAndRewriteExecLine(s *snap.Info, desktopFile, line string) (*snap.AppInfo, string, error) {
```

Comment:
nitpick: since you only actually use `app.Name`, we could consider just returning the name of the app (instead of returning pointers), and then use named return values
```suggestion
func detectAppAndRewriteExecLine(s *snap.Info, desktopFile, line string) (appName, execLine string, err error) {
```

Reply:
updated, thanks!